### PR TITLE
ENH: Exposes new transpose method

### LINKED ISSUE #195: New method: transpose
**Addition Description**
A new method should be added, `transpose`. This will accept a feature table as input and return the transposed feature table as output. This, coupled with the ability to view the sample axis of a `FeatureTable` as `Metadata` will allow users to work with both axes as `Metadata`.

**References**
1. https://github.com/qiime2/q2-types/pull/186

### PR DESCRIPTION
**Pull request by turanoo**
This method allows for transposing rows and columns in a feature table. 
Fixes #195

### CODE REVIEW

File: q2_feature_table/__init__.py [hunk 1]
```diff
@@ -20,8 +20,7 @@
 __version__ = get_versions()['version']
 del get_versions
 
-__all__ = ['rarefy', 'presence_absence', 'relative_frequency', 'summarize',
-           'merge', 'merge_seqs', 'filter_samples', 'filter_features',
-           'merge_taxa', 'tabulate_seqs', 'overlap_methods',
-           'core_features', 'group', 'heatmap', 'heatmap_choices',
-           'filter_seqs', 'subsample']
+__all__ = ['rarefy', 'presence_absence', 'relative_frequency', 'transpose',
+            'summarize','merge', 'merge_seqs', 'filter_samples', 'filter_features',
```
Comment (thermokarst) [inline q2_feature_table/__init__.py]: Lint error (extra space)

File: q2_feature_table/_transform.py [hunk 2]
```diff
@@ -21,3 +21,7 @@ def relative_frequency(table: biom.Table) -> biom.Table:
     """
     table.norm(axis='sample', inplace=True)
     return table
+
+def transpose(table: biom.Table) -> biom.Table:
+    table = table.transpose()
```
Comment (thermokarst) [inline q2_feature_table/_transform.py]: You might want to name this something like `transposed_table` - that way you aren't overwriting your input `table`.

File: q2_feature_table/plugin_setup.py [hunk 3]
```diff
@@ -117,6 +117,24 @@
...
+               FeatureTable[Frequency])],
+    input_descriptions={
+        'table': 'The feature table to be transposed.'
+    },
+    parameter_descriptions={},
+    output_descriptions={
+        'transposed_feature_table': ('The resulting transposed feature table.')
+    },
+    name='Transpose a feature table.',
+    description='Transpose the rows and columns in a feature table.'
```
Comment (thermokarst) [inline q2_feature_table/plugin_setup.py]: How about you replace `in` with `of`?

> Transpose the rows and columns of a feature table

File: q2_feature_table/__init__.py (hunk 1 above)
Comment (turanoo) [reply q2_feature_table/__init__.py]: Thanks for the review, I will take care of these!

File: q2_feature_table/plugin_setup.py [hunk 4]
```diff
@@ -117,6 +117,24 @@
...
+               FeatureTable[Frequency])],
+    input_descriptions={
+        'table': 'The feature table to be transposed.'
+    },
+    parameter_descriptions={},
+    output_descriptions={
+        'transposed_feature_table': ('The resulting transposed feature table.')
+    },
+    name='Transpose a feature table.',
+    description='Transpose the rows and columns of a feature table.'
```
Comment (thermokarst) [inline q2_feature_table/plugin_setup.py]: I was stewing on this a bit - what do you think of:

> Transpose the rows and columns (samples and features) of a feature table.

Comment (ebolyen) [reply q2_feature_table/plugin_setup.py]: :+1: maybe add: `(typically samples and features)` in the parenthetical?

**Comment by thermokarst** [comment]
Hey @turanoo, looks like you might need to re-lint this package using flake8 3.6 (this is being pulled in now that this package is being tested with python 3.6, it seems).

**Comment by thermokarst** [comment]
@turanoo - can you please uncommit `q2_feature_table/_summarize/summarize_assets/package-lock.json`?

**Pull request merged**